### PR TITLE
More flexible deploy-scripts for other deployments

### LINKED ISSUE #279: Zugriff auf Qualix während Deployment einschränken / Maintenance Mode

### PR DESCRIPTION
**Pull request by cleverer**
I'm addressing 3 issues with this PR:
- reenable ssh strict host checking. @carlobeltrame I actually got it working by just copying the hashed ssh-keyscan output form a failed run. The only difference seems to be the order of the keys and their hashed hostnames. But the keys themselves are the same as when I do the keyscan locally…
- ~~I removed the failing php version check, as this does not work with cyon, as they have a different php version for the bare `php` cli command and the actual running php version.~~  
    I changed the php version check error message to include a hint on how one might solve the cli problem… 
- I adjusted the CI Trigger action step, so it doesn't fail if no `REPO_ACCESS_TOKEN` is configured…
- Added some commands to the deploy script to put qualix in maintenance mode automatically (resolves #279)
- I adjusted the sentry action step, so it doesn't fail if no `env.SENTRY_AUTH_TOKEN` is configured…
- Fixed typo in deploy script, which prevented `APP_CONTACT_TEXT` from working
- Allow to set `APP_NAME` in CI

### CODE REVIEW

File: .github/actions/deploy/action.yml [hunk 1]
```diff
@@ -16,6 +16,11 @@ inputs:
     required: false
     default: .
 
+  app-name:
+    description: 'The App Name'
+    required: false
+    default: ''
```
Comment (carlobeltrame) [inline .github/actions/deploy/action.yml:22]: Default "Qualix"? Wir haben auch im .env.example Qualix drin.

**Review by carlobeltrame — APPROVED** [review]
Sieht alles super aus, merci vielmal. Ich habe noch eine Maintenance Page mit etwas mehr Infos für User gemacht, siehe https://github.com/pfadi-region-winterthur/qualix/pull/1

File: .github/actions/deploy/deploy.sh [hunk 2]
```diff
@@ -56,17 +60,24 @@ docker-compose run --no-deps --entrypoint "/bin/sh -c 'npm install && npm run pr
...
   php -v
   cd $SSH_DIRECTORY
-  php -r "if(PHP_VERSION_ID<${PHP_MIN_VERSION_ID:-80100}){echo \"Your PHP version is too old\\n\";exit(1);}"
+  php -r "if(PHP_VERSION_ID<${PHP_MIN_VERSION_ID:-80100}){echo \"Your PHP version is too old\\nYou might be able to use these instructions on your hosting as well: https://www.cyon.ch/support/a/php-standardversion-fur-die-kommandozeile-festlegen\n\";exit(1);}"
 EOF
 
-cat ~/.ssh/known_hosts
+ssh -l $SSH_USERNAME -T $SSH_HOST <<EOF
+  cd $SSH_DIRECTORY
+  php artisan down
```
Comment (carlobeltrame) [inline .github/actions/deploy/deploy.sh:79]: Könnte man das vielleicht gleich in der gleichen SSH Session machen wie der vorangehende PHP Versions-Check?

File: .github/actions/deploy/action.yml (hunk 1 above)
Comment (cleverer) [reply .github/actions/deploy/action.yml:22]: Hab das hier extra leer gelassen… wenns leer ist überschreibt das Deploy-Script das .env.example nicht… aber wenn du meinst, es ist klarer, wenns hier einen Default gibt, kann ich den auch setzen, dann nehm ich aber das if statement im script weg und machs required…

File: .github/actions/deploy/deploy.sh (hunk 2 above)
Comment (cleverer) [reply .github/actions/deploy/deploy.sh:79]: Ah ja, könnte man ^^

Kommt davon, dass ich zuerst noch das `cat ~/.ssh/known_hosts` für das ssh debugging drin hatte…

Comment (cleverer) [reply .github/actions/deploy/deploy.sh]: Habs amended…

File: .github/actions/deploy/action.yml (hunk 1 above)
Comment (carlobeltrame) [reply .github/actions/deploy/action.yml:22]: Ah, sorry, du hast völlig recht. Ich hatte das `if` im Deploy Skript vergessen. Ist okay so.

**Comment by carlobeltrame** [comment]
I just wanted to deploy this to a test server... Where did you set the known_hosts in your deployment tests which succeeded?

**Comment by cleverer** [comment]
> I just wanted to deploy this to a test server... Where did you set the known_hosts in your deployment tests which succeeded?

The `shimataro/ssh-key-action@v2` is already set up to consume the `SSH_KNOWN_HOSTS` secret…

**Comment by carlobeltrame** [comment]
Ah, thanks for explaining my own workflow for me :P will try afterwards. Did you just add all 3 of the keys which ssh-keyscan outputs? Or a specific one according to some criterion?

**Comment by carlobeltrame** [comment]
I added only the key which I also have on my machine. SSH login was successful. But for the first deployment, you need to manually upload the "updating.blade.php" view - it can't activate maintenance mode without having this view, and it can't upload this view without activating maintenance mode :P

**Comment by cleverer** [comment]
> I added only the key which I also have on my machine. SSH login was successful. But for the first deployment, you need to manually upload the "updating.blade.php" view - it can't activate maintenance mode without having this view, and it can't upload this view without activating maintenance mode :P

Fair point ^^
However, there is not much documentation for the CI deployment anyway ;)

**Pull request merged**